lib/active_job/retry.rb

```diff
@@ -117,6 +125,18 @@ def check_adapter!
       end
     end
 
+    def validate_params
+      if retry_callback && !retry_callback.is_a?(Proc)
```

Comment:
Ah, yeah good point. I wonder how valuable having it instance_exec'd would be - do you have an example usage where it's necessary?